cmake/modules/FindKOKKOS.cmake

```diff
@@ -0,0 +1,75 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2021 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE.md at
+## the top level directory of deal.II.
+##
+## ---------------------------------------------------------------------
+
+#
+# Try to find the Kokkos library
+#
+# This module exports
+#
+#   KOKKOS_INCLUDE_DIRS
+#   KOKKOS_INTERFACE_LINK_FLAGS
+#
+
+SET(KOKKOS_DIR "" CACHE PATH "An optional hint to a Kokkos installation")
+SET_IF_EMPTY(KOKKOS_DIR "$ENV{KOKKOS_DIR}")
+
+
+FIND_PACKAGE(Kokkos
+  HINTS ${KOKKOS_DIR} ${Kokkos_DIR} $ENV{Kokkos_DIR}
+  )
+
+IF(Kokkos_FOUND)
+  # We are only interested in Kokkos if it is not part of Trilinos
+  IF(TARGET Kokkos::kokkos)
+    GET_PROPERTY(KOKKOS_INSTALL_INCLUDE_DIR TARGET Kokkos::kokkos PROPERTY INTERFACE_INCLUDE_DIRECTORIES)
+    GET_PROPERTY(KOKKOS_EXTRA_LD_FLAGS_FULL TARGET Kokkos::kokkoscore PROPERTY INTERFACE_LINK_OPTIONS)
+    STRING(LENGTH "${KOKKOS_EXTRA_LD_FLAGS_FULL}" str_length_ld)
+    MATH(EXPR length_ld "${str_length_ld} - 24")
+    IF(${length_ld} GREATER 0)
+      STRING(SUBSTRING "${KOKKOS_EXTRA_LD_FLAGS_FULL}" 23 ${length_ld} KOKKOS_EXTRA_LD_FLAGS)
+    ENDIF()
+    GET_PROPERTY(KOKKOS_COMPILE_FLAGS_FULL TARGET Kokkos::kokkoscore PROPERTY INTERFACE_COMPILE_OPTIONS)
+    STRING(LENGTH "${KOKKOS_COMPILE_FLAGS_FULL}" str_length_c)
+    MATH(EXPR length_c "${str_length_c} - 27")
+    IF(${length_c} GREATER 0)
+      STRING(SUBSTRING "${KOKKOS_COMPILE_FLAGS_FULL}" 26 ${length_c} KOKKOS_COMPILE_FLAGS)
+      ADD_FLAGS(DEAL_II_CXX_FLAGS ${KOKKOS_COMPILE_FLAGS})
+    ENDIF()
```

Comment:
> For me, INTERFACE_LINK_OPTIONS is -fopenmp so that KOKKOS_EXTRA_LD_FLAGS becomes empty. Is that what we want?

Are you sure about that? My `INTERFACE_LINK_OPTIONS` is `"\$<\$<LINK_LANGUAGE:CXX>:-DKOKKOS_DEPENDENCE;-fopenmp=libomp>"` Why don't you have `-DKOKKOS_DEPENDENCE`? As far as I can tell the flag is always set.

> Wouldn't it be better to use a regex for that?

Sure but how? I need to get ride of language metadata to get the flags. Which is what this code does.

Reply:
> Sure but how? I need to get ride of language metadata to get the flags. Which is what this code does.

The regex below works for me for `INTERFACE_COMPILE_OPTIONS` so an according one should also work here.

> Are you sure about that? My INTERFACE_LINK_OPTIONS is "\$<\$<LINK_LANGUAGE:CXX>:-DKOKKOS_DEPENDENCE;-fopenmp=libomp>" Why don't you have -DKOKKOS_DEPENDENCE? As far as I can tell the flag is always set.

In the `develop` branch we removed it and reintroduced it again. So the version I tested with didn't have `KOKKOS_DEPENDENCE`. My point is just that we can't rely on anything specific.